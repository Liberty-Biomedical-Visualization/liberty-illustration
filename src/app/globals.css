@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --accent-50: 224 247 252;
  --accent-100: 178 224 240;
  --accent-200: 128 200 232;
  --accent-300: 77 176 215;
  --accent-400: 26 153 206;
  --accent-500: 3 152 206;
  --accent-600: 2 128 165;
  --accent-700: 2 109 136;
  --accent-800: 2 90 107;
  --accent-900: 2 61 77;
  --accent-950: 1 44 58;
  --background: 255 255 255;
  --foreground: 23 23 23;
}

body {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}
