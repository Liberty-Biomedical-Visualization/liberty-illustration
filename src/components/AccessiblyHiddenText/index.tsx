export default function AccessiblyHiddenText(props: AccessiblyHiddenTextProps) {
  const { children } = props;
  const className =
    "absolute border-0 h-px m-0 overflow-hidden p-0 w-px whitespace-nowrap";
  return <span className={className}>{children}</span>;
}

export interface AccessiblyHiddenTextProps {
  children: string;
}
