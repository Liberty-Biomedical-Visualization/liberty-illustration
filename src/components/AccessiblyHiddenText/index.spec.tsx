import { describe, expect, it } from "@jest/globals";
import { isInaccessible, render, screen } from "@testing-library/react";

import AccessiblyHiddenText from ".";

describe("AccessiblyHiddenText", () => {
  const children = "This is some accessible but visually hidden text.";

  it("should display passed children", () => {
    render(<AccessiblyHiddenText>{children}</AccessiblyHiddenText>);
    const result = screen.getByText(children);
    expect(result).toBeVisible();
  });

  it("should be accessible", () => {
    render(<AccessiblyHiddenText>{children}</AccessiblyHiddenText>);
    const element = screen.getByText(children);
    const result = isInaccessible(element);
    expect(result).toBe(false);
  });

  it("should match the snapshot", () => {
    const result = render(
      <AccessiblyHiddenText>{children}</AccessiblyHiddenText>,
    );
    expect(result.container).toMatchSnapshot();
  });
});
