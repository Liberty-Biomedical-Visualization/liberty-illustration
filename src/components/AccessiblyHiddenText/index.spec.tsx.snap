// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessiblyHiddenText should match the snapshot 1`] = `
<div>
  <span
    class="absolute border-0 h-px m-0 overflow-hidden p-0 w-px whitespace-nowrap"
  >
    This is some accessible but visually hidden text.
  </span>
</div>
`;
